test/integration/singlecluster/scheduler/excluderesources/scheduler_test.go

```diff
@@ -0,0 +1,196 @@
+		gomega.Expect(wl.Status.Admission).NotTo(gomega.BeNil())
+		resourceUsage := wl.Status.Admission.PodSetAssignments[0].ResourceUsage
+		gomega.Expect(resourceUsage).To(gomega.HaveKey(corev1.ResourceCPU))
+		gomega.Expect(resourceUsage).NotTo(gomega.HaveKey("networking.example.com/vpc"))
+		gomega.Expect(resourceUsage).NotTo(gomega.HaveKey("storage.example.com/disk"))
+	})
+
+	ginkgo.It("should enforce quota for non-excluded resources", func() {
+		ginkgo.By("Creating a workload that consumes most of the CPU quota")
+		wl1 := utiltestingapi.MakeWorkload("quota-wl-1", ns.Name).
+			Queue(kueue.LocalQueueName(lq.Name)).
+			Request(corev1.ResourceCPU, "8").
+			Request("networking.example.com/vpc", "1000").
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl1)).To(gomega.Succeed())
+
+		wl1Key := types.NamespacedName{Name: wl1.Name, Namespace: ns.Name}
+
+		ginkgo.By("Waiting for first workload to be admitted")
+		gomega.Eventually(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, wl1Key, wl1)).To(gomega.Succeed())
+			g.Expect(workload.IsAdmitted(wl1)).To(gomega.BeTrue())
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+		ginkgo.By("Creating a second workload that would exceed CPU quota")
+		wl2 := utiltestingapi.MakeWorkload("quota-wl-2", ns.Name).
+			Queue(kueue.LocalQueueName(lq.Name)).
+			Request(corev1.ResourceCPU, "5").
+			Request("networking.example.com/vpc", "1").
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl2)).To(gomega.Succeed())
+
+		wl2Key := types.NamespacedName{Name: wl2.Name, Namespace: ns.Name}
+
+		ginkgo.By("Verifying second workload is not admitted due to CPU quota")
+		gomega.Consistently(func() bool {
+			gomega.Expect(k8sClient.Get(ctx, wl2Key, wl2)).To(gomega.Succeed())
+			return workload.IsAdmitted(wl2)
+		}, util.ConsistentDuration, util.Interval).Should(gomega.BeFalse())
+	})
+
+	ginkgo.It("should use exact prefix matching", func() {
+		ginkgo.By("Creating a workload with a resource that doesn't match the excluded prefix")
+		wl := utiltestingapi.MakeWorkload("prefix-test-wl", ns.Name).
+			Queue(kueue.LocalQueueName(lq.Name)).
+			Request(corev1.ResourceCPU, "1").
+			Request("networking.other.com/vpc", "100").
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+		wlKey := types.NamespacedName{Name: wl.Name, Namespace: ns.Name}
+
+		ginkgo.By("Verifying workload is admitted")
+		gomega.Eventually(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, wlKey, wl)).To(gomega.Succeed())
+			g.Expect(workload.IsAdmitted(wl)).To(gomega.BeTrue())
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+		ginkgo.By("Verifying non-matching resource is NOT excluded (appears in resource usage)")
+		gomega.Expect(k8sClient.Get(ctx, wlKey, wl)).To(gomega.Succeed())
+		resourceUsage := wl.Status.Admission.PodSetAssignments[0].ResourceUsage
+		gomega.Expect(resourceUsage).To(gomega.HaveKey(corev1.ResourceName("networking.other.com/vpc")))
```

Comment:
Not entirely sure why but when I made the change to drop corev1.ResourceName here the integration test was failing.

Had to revert the change here but I pushed @mbobrovskyi requests everywhere else.